fails to build as a subproject (samples/paho_cs_sub.c:20:24: fatal error: MQTTClient.h: No such file or directory)

I have a very simple setup:

```
paho.mqtt.c     <-- fresh unmodified git clone (master)
CMakeLists.txt  <-- my cmake file
```
Contents of CMakeLists.txt:

```
cmake_minimum_required(VERSION 3.5.1)

project(myproject CXX C)

SET(PAHO_WITH_SSL            TRUE  CACHE BOOL "")
SET(PAHO_BUILD_STATIC        TRUE  CACHE BOOL "")
SET(PAHO_BUILD_DOCUMENTATION FALSE CACHE BOOL "")
SET(PAHO_BUILD_SAMPLES       TRUE  CACHE BOOL "")
SET(PAHO_BUILD_DEB_PACKAGE   FALSE CACHE BOOL "")
SET(PAHO_ENABLE_TESTING      TRUE  CACHE BOOL "")

add_subdirectory(paho.mqtt.c   build_paho-mqtt-c)
```

Now when I run cmake on this directory, and later make, it fails:

```
[...]
Scanning dependencies of target paho_cs_sub
[ 56%] Building C object build_paho-mqtt-c/src/samples/CMakeFiles/paho_cs_sub.dir/paho_cs_sub.c.o
[...]/tmp/paho-test/5/paho.mqtt.c/src/samples/paho_cs_sub.c:20:24: fatal error: MQTTClient.h: No such file or directory
compilation terminated.
build_paho-mqtt-c/src/samples/CMakeFiles/paho_cs_sub.dir/build.make:62: recipe for target 'build_paho-mqtt-c/src/samples/CMakeFiles/paho_cs_sub.dir/paho_cs_sub.c.o' failed
make[2]: *** [build_paho-mqtt-c/src/samples/CMakeFiles/paho_cs_sub.dir/paho_cs_sub.c.o] Error 1
CMakeFiles/Makefile2:703: recipe for target 'build_paho-mqtt-c/src/samples/CMakeFiles/paho_cs_sub.dir/all' failed
make[1]: *** [build_paho-mqtt-c/src/samples/CMakeFiles/paho_cs_sub.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```

I investigated the generated build.make, and it looks $C_INCLUDE pathset is incorrect. Why?

It builds fine without the samples (PAHO_BUILD_SAMPLES left out).

--- a/src/samples/CMakeLists.txt
+++ b/src/samples/CMakeLists.txt
@@ -20,8 +20,8 @@
 ## compilation/linkage settings
 INCLUDE_DIRECTORIES(
     .
-    ${CMAKE_SOURCE_DIR}/src
-    ${CMAKE_BINARY_DIR}
+    ${PROJECT_SOURCE_DIR}/src
+    ${PROJECT_BINARY_DIR}
     )
 
 IF (WIN32)

